src/Microsoft.TemplateEngine.Cli/CliTemplateInfo.cs

```diff
@@ -96,7 +96,7 @@ internal CliTemplateInfo(ITemplateInfo templateInfo, HostSpecificTemplateData cl
 
         internal bool IsHidden => _cliData.IsHidden;
 
-        internal IEnumerable<CliTemplateParameter> CliParameters => _parameters;
+        internal IReadOnlyDictionary<string, CliTemplateParameter> CliParameters => _parameters.ToDictionary(p => p.Name, p => p);
```

Comment:
good catch, actually using dictionary initially simplifies logic, and we don't need list.
About lazy: made it lazy now, however I'm not sure if we want to handle same parameters as error or not. 
If yes, it might be better to generate dictionary in constructor to catch error earlier.